### PR TITLE
Update React to 18

### LINKED ISSUE #49: Update to react 18
- [x] Update s-pipes-editor-ui to react 18
     - [x]  Update all libraries that depend on React, and remove any that do not support React 18.
     - [x] Find alternative libraries for any that are removed, and replace them in the application.

### PR DESCRIPTION
**Pull request by palagdan**
Resolves #49

### CODE REVIEW

**Comment by palagdan** [comment]
@blcham

Finally, the installation can be done without using --legacy-peer-deps. I updated some dependencies:
"semantic-ui-react": "^0.88.2" -> "semantic-ui-react": "^2.1.5"
"react-imported-component": "6.2.1" -> "react-imported-component": "^6.5.4"

I removed react-hot-loader because it has been replaced by Fast Refresh, as noted [here](https://www.npmjs.com/package/react-hot-loader). react-hot-loader was used for re-rendering components without refreshing the page in development mode, preserving states, refs, etc. I tried to configure Fast Refresh for Webpack, but it doesn't seem to work. I'll continue to troubleshoot this. However, if we migrate to Vite, I'm not sure if it's worth the effort.

Regarding the fork we found for replacing react-loadingg (a library for loading spinners), it's not acceptable because it randomly selects a loading spinner from the react-loadingg pool each time. Since I couldn't find any other suitable forks for this library, I created a custom spinner component that's quite similar to react-loadingg.

Additionally, I replaced react-delay-render (a library for delaying the rendering of a React component by a specified time in milliseconds) with my own custom component that performs the same function.

**Comment by palagdan** [comment]
@blcham 
I renamed all components from .js to .jsx because Vite requires the .jsx extension for JSX processing. The reason for this requirement is that, in most cases, plain .js files don’t need full AST transforms to work in the browser. Allowing JSX in .js files would mean every served file must undergo full AST processing, just in case it contains JSX

File: src/components/App.jsx [hunk 1]
```diff
@@ -2,22 +2,22 @@ import React, { Component } from "react";
 import { Switch, BrowserRouter as Router, Route } from "react-router-dom";
 import importedComponent from "react-imported-component";
 
-import Home from "./Home";
-import Loading from "./Loading";
+import Home from "./Home.jsx";
```
Comment (blcham) [inline src/components/App.jsx:5]: it is strange that u specify suffix, add it at least to TODO the list after vite migration

**Review by blcham — APPROVED** [review]
rebase and merge + create TODO list of what needs to be done after Vite migration

File: src/components/App.jsx (hunk 1 above)
Comment (palagdan) [reply src/components/App.jsx:5]: It was created by IDEA, not by me, but it works. I'll add it to the list.

**Pull request merged**